packages/tag/src/react/index.tsx

```diff
@@ -0,0 +1,143 @@
+import { compose, css } from 'glamor'
+import React, { HTMLAttributes } from 'react'
+
+import Halo from '@pluralsight/ps-design-system-halo'
+import { sizes as iconSizes } from '@pluralsight/ps-design-system-icon'
+import {
+  useTheme,
+  names as themeNames
+} from '@pluralsight/ps-design-system-theme'
+
+import stylesheet from '../css'
+import * as vars from '../vars'
+import { ValueOf } from '@pluralsight/ps-design-system-util'
+
+const styles = {
+  tag: ({
+    themeName,
+    clickable,
+    icon,
+    isPressed,
+    size
+  }: {
+    themeName: ValueOf<typeof themeNames>
+    clickable: boolean
+    icon: boolean
+    isPressed: boolean
+    size: ValueOf<typeof vars.sizes>
+  }) => {
+    const label = 'psds-tag'
+
+    return compose(
+      css(stylesheet[`.${label}`]),
+      css(stylesheet[`.${label}.psds-theme--${themeName}`]),
+      clickable && css(stylesheet[`.${label}--clickable`]),
+      clickable &&
+        css(stylesheet[`.${label}--clickable.psds-theme--${themeName}`]),
+      css(stylesheet[`.${label}--size-${size}`]),
+      icon && css(stylesheet[`.${label}--icon`]),
+      isPressed && css(stylesheet[`.${label}--isPressed`]),
+      isPressed &&
+        clickable &&
+        css(stylesheet[`.${label}--isPressed.${label}--clickable`])
+    )
+  },
+  label: (themeName: ValueOf<typeof themeNames>, icon: boolean) => {
+    const label = 'psds-tag__label'
+
+    return compose(
+      css(stylesheet[`.${label}`]),
+      css(stylesheet[`.${label}.psds-theme--${themeName}`]),
+      icon && stylesheet[`.${label}--icon`]
+    )
+  }
+}
+
+export interface TagStatics {
+  sizes: typeof vars.sizes
+}
+
+export interface TagProps
+  extends HTMLAttributes<HTMLAnchorElement & HTMLDivElement> {
```

Comment:
Unlike Star, I think this is insufficient and should be updated to be polymorphic.

This is because one of the main goals of types is to give proper introspection into our components so devs will use them correctly.  

`<Tag download />` shouldn't be legal.

`<Tag href="file.zip" download />` should be.

But both are with this typing.

Reply:
sorta fixed I really think we need to figure this one out